File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/RelativeDates.cpp
// RelativeDates.cpp 
#include "RelativeDates.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

RelativeDates::RelativeDates(TiXmlNode* xmlNode)
: RelativeDateOffset(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //periodSkipNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* periodSkipNode = xmlNode->FirstChildElement("periodSkip");

   if(periodSkipNode){periodSkipIsNull_ = false;}
   else{periodSkipIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- periodSkipNode , address : " << periodSkipNode << std::endl;
   #endif
   if(periodSkipNode)
   {
      if(periodSkipNode->Attribute("href") || periodSkipNode->Attribute("id"))
      {
          if(periodSkipNode->Attribute("id"))
          {
             periodSkipIDRef_ = periodSkipNode->Attribute("id");
             periodSkip_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(periodSkipNode));
             periodSkip_->setID(periodSkipIDRef_);
             IDManager::instance().setID(periodSkipIDRef_,periodSkip_);
          }
          else if(periodSkipNode->Attribute("href")) { periodSkipIDRef_ = periodSkipNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { periodSkip_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(periodSkipNode));}
   }

   //scheduleBoundsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* scheduleBoundsNode = xmlNode->FirstChildElement("scheduleBounds");

   if(scheduleBoundsNode){scheduleBoundsIsNull_ = false;}
   else{scheduleBoundsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- scheduleBoundsNode , address : " << scheduleBoundsNode << std::endl;
   #endif
   if(scheduleBoundsNode)
   {
      if(scheduleBoundsNode->Attribute("href") || scheduleBoundsNode->Attribute("id"))
      {
          if(scheduleBoundsNode->Attribute("id"))
          {
             scheduleBoundsIDRef_ = scheduleBoundsNode->Attribute("id");
             scheduleBounds_ = boost::shared_ptr<DateRange>(new DateRange(scheduleBoundsNode));
             scheduleBounds_->setID(scheduleBoundsIDRef_);
             IDManager::instance().setID(scheduleBoundsIDRef_,scheduleBounds_);
          }
          else if(scheduleBoundsNode->Attribute("href")) { scheduleBoundsIDRef_ = scheduleBoundsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { scheduleBounds_ = boost::shared_ptr<DateRange>(new DateRange(scheduleBoundsNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypePositiveInteger> RelativeDates::getPeriodSkip()
{
   if(!this->periodSkipIsNull_){
        if(periodSkipIDRef_ != ""){
             return boost::shared_static_cast<XsdTypePositiveInteger>(IDManager::instance().getID(periodSkipIDRef_));
        }else{
             return this->periodSkip_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypePositiveInteger>();
   }
}
boost::shared_ptr<DateRange> RelativeDates::getScheduleBounds()
{
   if(!this->scheduleBoundsIsNull_){
        if(scheduleBoundsIDRef_ != ""){
             return boost::shared_static_cast<DateRange>(IDManager::instance().getID(scheduleBoundsIDRef_));
        }else{
             return this->scheduleBounds_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<DateRange>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/RelativeDates.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class RelativeDates : RelativeDateOffset
    {
        public RelativeDates(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode periodSkipNode = xmlNode.SelectSingleNode("periodSkip");
            
            if (periodSkipNode != null)
            {
                if (periodSkipNode.Attributes["href"] != null || periodSkipNode.Attributes["id"] != null) 
                {
                    if (periodSkipNode.Attributes["id"] != null) 
                    {
                        periodSkipIDRef_ = periodSkipNode.Attributes["id"].Value;
                        XsdTypePositiveInteger ob = new XsdTypePositiveInteger(periodSkipNode);
                        IDManager.SetID(periodSkipIDRef_, ob);
                    }
                    else if (periodSkipNode.Attributes["href"] != null)
                    {
                        periodSkipIDRef_ = periodSkipNode.Attributes["href"].Value;
                    }
                    else
                    {
                        periodSkip_ = new XsdTypePositiveInteger(periodSkipNode);
                    }
                }
                else
                {
                    periodSkip_ = new XsdTypePositiveInteger(periodSkipNode);
                }
            }
            
        
            XmlNode scheduleBoundsNode = xmlNode.SelectSingleNode("scheduleBounds");
            
            if (scheduleBoundsNode != null)
            {
                if (scheduleBoundsNode.Attributes["href"] != null || scheduleBoundsNode.Attributes["id"] != null) 
                {
                    if (scheduleBoundsNode.Attributes["id"] != null) 
                    {
                        scheduleBoundsIDRef_ = scheduleBoundsNode.Attributes["id"].Value;
                        DateRange ob = new DateRange(scheduleBoundsNode);
                        IDManager.SetID(scheduleBoundsIDRef_, ob);
                    }
                    else if (scheduleBoundsNode.Attributes["href"] != null)
                    {
                        scheduleBoundsIDRef_ = scheduleBoundsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        scheduleBounds_ = new DateRange(scheduleBoundsNode);
                    }
                }
                else
                {
                    scheduleBounds_ = new DateRange(scheduleBoundsNode);
                }
            }
            
        
        }
        
    
        #region PeriodSkip_
        private XsdTypePositiveInteger periodSkip_;
        public XsdTypePositiveInteger PeriodSkip_
        {
            get
            {
                if (this.periodSkip_ != null)
                {
                    return this.periodSkip_; 
                }
                else if (this.periodSkipIDRef_ != null)
                {
                    periodSkip_ = IDManager.getID(periodSkipIDRef_) as XsdTypePositiveInteger;
                    return this.periodSkip_; 
                }
                else
                {
                      return this.periodSkip_; 
                }
            }
            set
            {
                if (this.periodSkip_ != value)
                {
                    this.periodSkip_ = value;
                }
            }
        }
        #endregion
        
        public string periodSkipIDRef_ { get; set; }
        #region ScheduleBounds_
        private DateRange scheduleBounds_;
        public DateRange ScheduleBounds_
        {
            get
            {
                if (this.scheduleBounds_ != null)
                {
                    return this.scheduleBounds_; 
                }
                else if (this.scheduleBoundsIDRef_ != null)
                {
                    scheduleBounds_ = IDManager.getID(scheduleBoundsIDRef_) as DateRange;
                    return this.scheduleBounds_; 
                }
                else
                {
                      return this.scheduleBounds_; 
                }
            }
            set
            {
                if (this.scheduleBounds_ != value)
                {
                    this.scheduleBounds_ = value;
                }
            }
        }
        #endregion
        
        public string scheduleBoundsIDRef_ { get; set; }
        
    
        
    
    }
    
}

